src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,37 @@
 'use strict';
-
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (target instanceof Herbivore && target.hidden !== true) {
```

Comment:
it's easier to check with just a logical `no` here
```suggestion
    if (target instanceof Herbivore && !target.hidden) {
```